pynest/examples/EI_clustered_network/helper.py

```diff
@@ -0,0 +1,185 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""PyNEST EI-clustered network: Helper Functions
+------------------------------------------------
+
+Helper functions to calculate synaptic weights to construct
+random balanced networks and plot raster plot with color groups.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+small = 1e-10
+
+
+def postsynaptic_current_to_potential(tau_m, tau_syn, c_m=1.0, e_l=0.0):
+    """Maximum post-synaptic potential amplitude
+    for exponential synapses and a synaptic efficacy J of 1 pA.
+
+    Parameters
+    ----------
+    tau_m: float
+        Membrane time constant [ms]
+    tau_syn: float
+        Synapse time constant  [ms]
+    c_m: float (optional)
+        Membrane capacity [pF] (default: 1.0)
+    e_l: float (optional)
+        Resting potential [mV] (default: 0.0)
+
+    Returns
+    -------
+    float
+        maximum psp amplitude (for a 1 pA current spike) [mV]
+    """
+    # time of maximum deflection of the psp  [ms]
+    tmax = np.log(tau_syn / tau_m) / (1 / tau_m - 1 / tau_syn)
+    # we assume here the current spike is 1 pA, otherwise [mV/pA]
+    pre = tau_m * tau_syn / c_m / (tau_syn - tau_m)
+    return (e_l - pre) * np.exp(-tmax / tau_m) + pre * np.exp(-tmax / tau_syn)
+
+
+def calculate_RBN_weights(params):
+    """Calculate synaptic weights for a random balanced network.
+
+    The synaptic weights are calculated according to the method
+    described in Eqs. 7-10 [1]_.
+
+    References
+    ----------
+    .. [1] Rostami V, Rost T, Riehle A, van Albada SJ and Nawrot MP. 2020.
+        Excitatory and inhibitory motor cortical clusters account for balance,
+        variability, and task performance. bioRxiv 2020.02.27.968339.
+        DOI: `10.1101/2020.02.27.968339
+        <https://doi.org/10.1101/2020.02.27.968339>`__.
+
+    Parameters
+    ----------
+    params: dict
+        Dictionary of network parameters
+
+    Returns
+    -------
+    ndarray
+        synaptic weights 2x2 matrix [[EE, EI], [IE, II]]
+    """
+    N_E = params.get("N_E")  # excitatory units
+    N_I = params.get("N_I")  # inhibitory units
+    N = N_E + N_I  # total units
+    baseline_conn_prob = params.get("baseline_conn_prob")  # connection probs
+    gei = params.get("gei")
+    gii = params.get("gii")
+    gie = params.get("gie")
+    V_th_E = params.get("V_th_E")  # threshold voltage
+    V_th_I = params.get("V_th_I")
+    tau_E = params.get("tau_E")
+    tau_I = params.get("tau_I")
+    E_L = params.get("E_L")
+    tau_syn_ex = params.get("tau_syn_ex")
+    tau_syn_in = params.get("tau_syn_in")
+    amp_EE = postsynaptic_current_to_potential(tau_E, tau_syn_ex)
+    amp_EI = postsynaptic_current_to_potential(tau_E, tau_syn_in)
+    amp_IE = postsynaptic_current_to_potential(tau_I, tau_syn_ex)
+    amp_II = postsynaptic_current_to_potential(tau_I, tau_syn_in)
```

Comment:
A few more empty lines and some re-ordering make the code more readable, I believe.. Why do you use `.get()` everywhere instead of `params["N_E"]` and similar? That would be more readable for most Python users.

```suggestion

    N_E = params.get("N_E")  # excitatory units
    N_I = params.get("N_I")  # inhibitory units
    N = N_E + N_I  # total units

    E_L = params.get("E_L")
    V_th_E = params.get("V_th_E")  # threshold voltage
    V_th_I = params.get("V_th_I")
    
    tau_E = params.get("tau_E")
    tau_I = params.get("tau_I")
    
    tau_syn_ex = params.get("tau_syn_ex")
    tau_syn_in = params.get("tau_syn_in")
    
    gei = params.get("gei")
    gii = params.get("gii")
    gie = params.get("gie")
    
    amp_EE = postsynaptic_current_to_potential(tau_E, tau_syn_ex)
    amp_EI = postsynaptic_current_to_potential(tau_E, tau_syn_in)
    amp_IE = postsynaptic_current_to_potential(tau_I, tau_syn_ex)
    amp_II = postsynaptic_current_to_potential(tau_I, tau_syn_in)    
    
    baseline_conn_prob = params.get("baseline_conn_prob")  # connection probs
```